src/selectors/per-thread/composed.js

```diff
@@ -76,6 +82,28 @@ export function getComposedSelectorsPerThread(
       if (!jsTracerTable) {
         visibleTabs = visibleTabs.filter((tabSlug) => tabSlug !== 'js-tracer');
       }
+      let hasSamples = samples.length > 0 && stackTable.length > 0;
+      if (hasSamples) {
+        const stackIndex = ensureExists(samples.stack[0]);
+        if (stackTable.prefix[stackIndex] === null) {
+          // There's only a single stack frame, check if it's '(root)'.
+          const frameIndex = stackTable.frame[stackIndex];
+          const funcIndex = frameTable.func[frameIndex];
+          const stringIndex = funcTable.name[funcIndex];
+          if (stringTable.getString(stringIndex) === '(root)') {
+            // If the first sample's stack is only the root, check if any other
+            // sample is different.
+            hasSamples = samples.stack.some((s) => s !== stackIndex);
```

Comment:
Hm, this looks a bit costly, but probably it's going to return early after two iterations for most of our profiles. And not sure how else we can do it without looping. I think it's okay to keep like this.

Reply:
it could be faster to:
1. get the string index for (root) outside of the loop
2. compare string indexes here

Reply:
The `.some` call (assuming that's what is being called 'loop' in these comments) should only be done for profiles where the first sample contains a single frame, and then we only compare one integer with an array of integers. That should be fast enough, and we will only compare the entire array of integers (one per sample) when the profile was captured with nostacksampling and all the sample contain only "(root)".